app/main.py

```diff
@@ -66,6 +65,12 @@ def setup_cors(self):
             allow_headers=["*"],
         )
 
+    def setup_data(self):
+        if os.getenv("isDevelopmentMode") == "enabled":
+            logging.info("Development mode enabled. Skipping Firebase setup.")
```

Comment:
ideally yes, since we disable GitHub auth, and parsing and further interactions with the project does require a user , so we should have a default user creation to actually let things happen further.

So @waveywaves maybe what should happen is that all the major apis auth should become optional, and if development mode is enabled, we should skip auth and use the default user. This default user should ideally be inserted in database (if not present) when app starts with development mode.

We do not need to go to each router and make auth optional but do it at a common place. I think this is critical to go forward.

@waveywaves let me know if you have questions

Reply:
> ideally yes, since we disable GitHub auth, and parsing and further interactions with the project does require a user , so we should have a default user creation to actually let things happen further.

yep, I see the default user being returned by the AuthService

> We do not need to go to each router and make auth optional but do it at a common place. I think this is critical to go forward.

check_auth which is used by all the routers is already returning a default user.

Reply:
I haven't really made any code changes to setup_data, I have just put the firebase init validation for isDevelopmentMode in the setup_data function. The AuthService already returns the default_user with check_auth for `ifDevelopmentMode=enabled`. What more changes do we need to make exactly ?

Reply:
yes I just saw that, that part has not been tested though. I think I will resolve this conversation and test the rest of pr out for approval.